src/main/java/io/vertx/core/json/Json.java

```diff
@@ -166,7 +167,7 @@ public static String encodePrettily(Object obj) throws EncodeException {
    */
   public static <T> T decodeValue(Buffer buf, Class<T> clazz) throws DecodeException {
     try {
-      return mapper.readValue(new ByteBufInputStream(buf.getByteBuf()), clazz);
+      return mapper.readValue((InputStream) new ByteBufInputStream(buf.getByteBuf()), clazz);
```

Comment:
The `ByteBufInputStream` is both a `InputStream` and a `DataReader`. Jackson `2.8.x` introduces overloads into `ObjectMapper` for `DataReader`. Therefore, this call is now ambiguous and the cast is needed to indicate which method to invoke. Alternatively, you could take a look variable as type `InputStream` (not `ByteBufInputStream`) to achieve the same result.